### PR TITLE
Search: add folders and subforders to search info

### LINKED ISSUE #92973: Search: Document sub folders not searching
Current nested folders search implementation has a known limitation that when searching in a folder, subfolders of that folder are not searched. The only way to search in subfolders is to search from the root view.

This known limitation should be documented to help people out.

### PR DESCRIPTION
**Pull request by eledobleefe**
**What is this feature?**
Modify the search documentation by adding `folders` and `subfolders` to it.

**Why do we need this feature?**
To inform users about a known issue when searching for subfolders within a folder.

**Who is this feature for?**
Everybody

**Which issue(s) does this PR fix?**:

Fixes #92973 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: docs/sources/search/_index.md [hunk 1]
```diff
@@ -13,28 +14,38 @@ title: Search
...
 {{% /admonition %}}
 
-The following image shows the search results when you search using dashboard name.
+The following image shows the search results when you search a dashboard by name.
 
 {{< figure src="/static/img/docs/v91/dashboard-features/search-by-dashboard-name.png" width="700px" >}}
 
+The following images shows the search of a folder and the search for a dashboard inside that folder.
+
+TODO >>>> add screenshots
```
Comment (eledobleefe) [inline docs/sources/search/_index.md:43]: I'll wait for the text to be approved to add the screenshots

File: docs/sources/search/_index.md [hunk 2]
```diff
@@ -13,28 +14,38 @@ title: Search
 weight: 80
 ---
 
-# Search dashboards
+# Search dashboards and folders
 
-You can search for dashboards by dashboard name and by panel title. When you search for dashboards, the system returns all dashboards available within the Grafana instance, even if you do not have permission to view the contents of the dashboard.
+You can search for dashboards and folders by name.
```
Comment (imatwawana) [inline docs/sources/search/_index.md:19]: I'd like to be explicit about this.
```suggestion
You can search for dashboards and dashboard folders by name.
```

File: docs/sources/search/_index.md [hunk 3]
```diff
@@ -13,28 +14,38 @@ title: Search
...
+The following image shows the search results when you search a dashboard by name.
 
 {{< figure src="/static/img/docs/v91/dashboard-features/search-by-dashboard-name.png" width="700px" >}}
 
+The following images shows the search of a folder and the search for a dashboard inside that folder.
+
+TODO >>>> add screenshots
+
+{{% admonition type="note" %}}
+When searching within a folder, the subfolders will not be return. You need to use the main search page to find them.
```
Comment (imatwawana) [inline docs/sources/search/_index.md:46]:
```suggestion
When you search within a folder, its subfolders are not part of the results returned. You need to be on the **Dashboards** page (or the root level) to search for subfolders by name.
```

File: docs/sources/search/_index.md [hunk 4]
```diff
@@ -13,28 +14,38 @@ title: Search
...
-You can use your keyboard arrow keys to navigate the results and press `Enter` to open the selected dashboard.
+You can use your keyboard arrow keys to navigate the results and press `Enter` to open the selected dashboard or folder.
 {{% /admonition %}}
 
-The following image shows the search results when you search using dashboard name.
+The following image shows the search results when you search a dashboard by name.
 
 {{< figure src="/static/img/docs/v91/dashboard-features/search-by-dashboard-name.png" width="700px" >}}
 
+The following images shows the search of a folder and the search for a dashboard inside that folder.
```
Comment (imatwawana) [inline docs/sources/search/_index.md:41]: The page starts out talking about searching for dashboards and folders by name, so maybe there should be three total examples:

- Searching by dashboard name from the Dashboards page
- Searching by folder name from the Dashboards page
- Searching by dashboard name inside a folder

**Review by imatwawana — COMMENTED** [review]
Great start! I've made some editing suggestions as well as some larger content suggestions.

File: docs/sources/search/_index.md (hunk 4 above)
Comment (eledobleefe) [reply docs/sources/search/_index.md:41]: Yes. The first one, '_Searching by dashboard name from the Dashboards page_', was already on the Search page. The following two, '_Searching by folder name from the Dashboards page_' and '_Searching by dashboard name inside a folder_', are the ones that I want to include. 
Do you want me to be more specific? For example:

> The following images show:
1.- Searching by dashboard name from the Dashboards page
|>>> Here the screenshot
2.- Searching by folder name from the Dashboards page
|>>> Screenshot
3.- Searching by dashboard name inside a folder
|>>> Screenshot

Comment (imatwawana) [reply docs/sources/search/_index.md:41]: Yes, exactly! I think just so it's really clear the two different starting points and what one is searching for.

File: docs/sources/search/_index.md [hunk 5]
```diff
@@ -13,28 +14,44 @@ title: Search
...
 
 {{% admonition type="note" %}}
-You can use your keyboard arrow keys to navigate the results and press `Enter` to open the selected dashboard.
+You can use your keyboard arrow keys to navigate the results and press `Enter` to open the selected dashboard or folder.
 {{% /admonition %}}
 
-The following image shows the search results when you search using dashboard name.
+The following images show:
+
+1. Searching by dashboard name from the Dashboards page.
```
Comment (imatwawana) [inline docs/sources/search/_index.md:39]:
```suggestion
Searching by dashboard name from the **Dashboards** page.
```

File: docs/sources/search/_index.md [hunk 6]
```diff
@@ -13,28 +14,44 @@ title: Search
...
 
 {{< figure src="/static/img/docs/v91/dashboard-features/search-by-dashboard-name.png" width="700px" >}}
 
+2. Searching by folder name from the Dashboards page.
```
Comment (imatwawana) [inline docs/sources/search/_index.md:43]:
```suggestion
Searching by folder name from the **Dashboards** page.
```

File: docs/sources/search/_index.md [hunk 7]
```diff
@@ -13,28 +14,44 @@ title: Search
...
+
+{{< figure src="/media/docs/grafana/dashboards/search-folder-by-name.png" width="700px" >}}
+
+3. Searching by dashboard name inside a folder.
```
Comment (imatwawana) [inline docs/sources/search/_index.md:47]:
```suggestion
Searching by dashboard name inside a folder.
```

**Review by imatwawana — COMMENTED** [review]
This is great! The only issue is that the detail in the images is really hard to see, which is because the images are so big:
image

I would recommend making your browser much smaller (in width) and retaking the images so that it's easier to see the image details without having to open them. And yes, unfortunately you'll have to name them slightly differently to upload them to the media folder. :-S

**Review by imatwawana — APPROVED** [review]
Looks awesome! Thank you so much for all your extra effort on this!

**Review by irenerl24 — APPROVED** [review]

**Review by irenerl24 — APPROVED** [review]

**Pull request merged**